doc/CONTRIBUTING.md

```diff
@@ -72,26 +72,46 @@ If you are not sure of what to do, don't hesitate to open an issue about your PR
 
 #### The commit message
 
-The commit message has to be clear and related to the PR content.
+Sonata is a big project with many contributors, and a big part of the job is
+being able to understand the code at all times, be it when submitting a PR or
+looking at the history. Good commit messages are crucial to achieve this goal.
 
-The first line of the commit message must be short.
-The other lines must contains a complete description of what you done and why.
+There are already a few articles (or even single purpose websites) about this,
+we cannot recommend enough the following :
 
-The description is optional but recommended. It could be asked by the team if needed.
+- http://rakeroutes.com/blog/deliberate-git
+- http://stopwritingramblingcommitmessages.com
+- http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
 
-Bad commit message:
+To sum them up, the commit message has to be crystal clear and of course,
+related to the PR content.
+
+The first line of the commit message must be short, keep it under 50
+characters. It must say concisely but *precisely* what you did. The other lines
+must contain a complete description of *why* you did this.
```

Comment:
The other lines, if needed, should contains...

In some case for very little modification, having a description is completely trivial.

Example: `Fixing typos on "Get started" documentation` no need more description IMHO.